crates/inspector/src/eip3155.rs

```diff
@@ -283,7 +283,7 @@ where
     fn call_end(&mut self, context: &mut CTX, _: &CallInputs, outcome: &mut CallOutcome) {
         self.gas_inspector.call_end(outcome);
 
-        if context.journal_mut().depth() == 0 {
+        if context.journal_mut().depth() == 1 {
```

Comment:
```suggestion
        if context.journal_mut().depth() == 0 {
```

This probably should be 0 , as call_end has decremented depth